TensorIO/tensorio/src/main/java/ai/doc/tensorio/TIOModel/TIOModelIO.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * TIOModelIO.java
+ * TensorIO
+ *
+ * Created by Philip Dow on 7/6/2020
+ * Copyright (c) 2020 - Present doc.ai (http://doc.ai)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ai.doc.tensorio.TIOModel;
+
+import android.support.annotation.Nullable;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.List;
+import java.util.Set;
+
+import ai.doc.tensorio.TIOLayerInterface.TIOLayerInterface;
+
+/**
+ * Encapsulates information about the inputs, outputs, and placeholders for a model.
+ */
+
+public class TIOModelIO {
+
+    /**
+     * An I/O list may be indexed by key or by index.
+     */
+
+    public class TIOModelIOList {
+
+        private List<TIOLayerInterface> indexedInterfaces;
+        private Map<String, TIOLayerInterface> namedInterfaces;
+        private Map<String, Integer> nameToIndex;
+
+        /**
+         * Initializes an indexed model list with a list interfaces. You should not
+         * need to create instances of this class yourself.
+         *
+         * If the initializing interfaces parameter is nil, it will be treated as an
+         * empty list.
```

Comment:
It'd be nice to have 
@param A list of TIOLayerInterface objects
added to the function description.

Similar for other functions.